src/control/provider/system/mocks.go

```diff
@@ -44,10 +44,19 @@ type (
 		SourceToTarget  map[string]string
 		GetfsIndex      int
 		GetfsUsageResps []GetfsUsageRetval
-		GetFsTypeRes    *FsType
-		GetFsTypeErr    []error
+		GetfsTypeRes    *FsType
+		GetfsTypeErr    []error
 		StatErrors      map[string]error
 		RealStat        bool
+		ReadFileResults map[string][]byte
+		ReadFileErrors  map[string]error
+		RealReadFile    bool
```

Comment:
I see `RealStat` already exists up there, but I don't feel great about adding this kind of complexity to a single mock struct, to toggle the real version of the function on and off. IMO, if there are certain tests that want some real and some mocked functions, those tests should create their own mock struct to customize the behavior.

Reply:
From my side, it has the advantage to factorize this mocking code and the complexity seems to still be acceptable.
However, as I am not a GO expert, I am probably missing something.

Reply:
@knard-intel those were my thoughts when adding the flag. After chatting with @kjacque I can see benefit in moving to smaller package-local (as @mjmac also suggested elsewhere) SystemProvider mocks which would be consistent with having package-level SystemProvider interfaces. I will make changes if I have to repush or the next time I visit this area.

Reply:
Make sense, thanks for the explanation :-)